lib/opencensus/trace/link.rb

```diff
@@ -0,0 +1,81 @@
+# Copyright 2017 OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module OpenCensus
+  module Trace
+    ##
+    # A pointer from the current span to another span in the same trace or in
+    # a different trace. For example, this can be used in batching operations,
+    # where a single batch handler processes multiple requests from different
+    # traces or when the handler receives a request from a different project.
+    #
+    class Link
+      # The relationship of the two spans is unknown, or known but other
+      # than parent-child.
+      LINK_TYPE_UNSPECIFIED         = :TYPE_UNSPECIFIED
```

Comment:
Name the constants the same as your current symbol values. The `LINK_TYPE` prefixes are a bit too verbose, and seem redundant given the class name is also "Link".